src/carousel/components/runtime.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useState } from "react";
+import { IframeRuntime } from "./iframe-runtime";
+import { IAuthoredState, IInteractiveState } from "./types";
+import { renderHTML } from "../../shared/utilities/render-html";
+import { Carousel } from "react-responsive-carousel";
+
+import "react-responsive-carousel/lib/styles/carousel.min.css"; // requires a loader
+import css from "./runtime.scss";
+
+interface IProps {
+  authoredState: IAuthoredState;
+  interactiveState?: IInteractiveState | null;
+  setInteractiveState?: (updateFunc: (prevState: IInteractiveState | null) => IInteractiveState) => void;
+}
+
+export const Runtime: React.FC<IProps> = ({ authoredState, interactiveState, setInteractiveState }) => {
+
+  const [currentSlide, setCurrentSlide] = useState(0);
+
+  const subinteractives = authoredState.subinteractives || [];
+  if (subinteractives.length === 0) {
+    return <div>No sub items available. Please add them using the authoring interface.</div>;
+  }
+  const levelsCount = subinteractives.length;
+
+  const currentSubintId = interactiveState?.currentSubinteractiveId;
+  let currentInteractive = subinteractives.find(si => si.id === currentSubintId);
+  if (!currentInteractive) {
+    currentInteractive = subinteractives[0];
+  }
+
+  const currentSubintIndex = subinteractives.indexOf(currentInteractive);
+  const currentLevel = levelsCount - currentSubintIndex;
+
+  const subStates = interactiveState?.subinteractiveStates;
+  const subState = subStates && subStates[currentInteractive.id];
+
+  const getAnswerText = (level: number, subinteractiveAnswerText: string | undefined) =>
+    `[Level: ${level}] ${subinteractiveAnswerText ? subinteractiveAnswerText : "no response"}`;
+
+  const handleNewInteractiveState = (interactiveId: string, newInteractiveState: any) => {
+    setInteractiveState?.((prevState: IInteractiveState) => {
+      const updatedStates = {...prevState?.subinteractiveStates, [interactiveId]: newInteractiveState };
+      return {
+        ...prevState,
+        answerType: "interactive_state",
+        subinteractiveStates: updatedStates,
+        answerText: getAnswerText(currentLevel, newInteractiveState.answerText)
+      };
+    });
+  };
+
+  const nextSlide = () => {
+    setCurrentSlide(currentSlide + 1);
+  };
+
+  const previousSlide = () => {
+    setCurrentSlide(currentSlide - 1);
+  };
+
+  // TODO: Instead of the below, make a pause API in lara-interactive-api
+  const pauseAllVideos = () => {
+    const iframes = document.querySelectorAll("iframe");
+    iframes.forEach( (iframe: any) => {
+      const video = iframe.contentWindow.document.querySelector("video");
+      video?.pause();
+    });
+  };
```

Comment:
This is a bit brutal but should work in practice before we have a proper API. :) Did you check if video-interactive recognizes that the video was paused in a programmatic way and updates its controls and state correctly?

Reply:
@pjanik I didn't check that, at least not completely. It's not immediately clear to me how to check the state, but I'll look into it. If you have any suggestions about how to check that, please let me know. For what it's worth, when looking at the carousel in runtime in a web browser, the videos do pause when they should and their controls behave just as you would expect them to when paused.

Reply:
> For what it's worth, when looking at the carousel in runtime in a web browser, the videos do pause when they should and their controls behave just as you would expect them to when paused.

That's nice, and I meant mostly that. Actually video interactive saves its state (e.g. progress) if I remember correctly, but I'm not sure how well that works in practice. One additional test would be to use a carousel so the video is paused, wait a bit, reload a page and see if the progress has been saved correctly (if video progress matches the moment when you used the carousel). But first I'd check if it works like that even without a carousel. ;)

Reply:
Yes, I tested standalone video interactives and they do save their progress. If you pause a video, reload the page, then press the video's play button, it starts off where you had paused it before.

The videos within the carousel behave the same way, even if they're paused using the brutal method :) So that's good.

One semi-related issue is that the carousel doesn't save its state. So if you're on the second item and reload the page, the carousel returns to the first item. I'm not sure what the desired behavior is there, but I'll ask.